core/src/mast/node/basic_block_node/arbitrary.rs

```diff
@@ -194,42 +198,214 @@ impl Default for MastForestParams {
+        // BasicBlockNode generation must reference decorator IDs in [0, decorators)
+        let bb_params = BasicBlockNodeParams {
+            max_decorator_id_u32: params.decorators,
+            ..Default::default()
+        };
+
+        // 1) Generate a Vec<BasicBlockNode> with length in `params.blocks`
+        (
+            prop::collection::vec(any_with::<BasicBlockNode>(bb_params), params.blocks.clone()),
+            prop::collection::vec(any::<Decorator>(), params.decorators as usize..=params.decorators as usize)
+        )
+            // 2) Map concrete blocks -> build a concrete MastForest
+            .prop_map(move |(blocks, decorators)| {
+                let mut forest = MastForest::new();
+
+                // Pre-populate the decorator ID space so referenced IDs are valid.
+                // Generate all decorator types for more comprehensive testing
+                for decorator in decorators {
+                    forest
+                        .add_decorator(decorator)
+                        .expect("Failed to add decorator");
+                }
+
+                // Insert the generated blocks into the forest
+                for block in blocks {
+                    forest.add_node(block).expect("Failed to add block");
+                }
 
-            // Pre-populate the decorator ID space so referenced IDs are valid.
-            // TODO: Replace Decorator::Trace(i) with Arbitrary for Decorator
-            for i in 0..params.decorators {
                 forest
-                    .add_decorator(Decorator::Trace(i))
-                    .expect("Failed to add decorator");
-            }
+            }).boxed()
+    }
+}
 
-            // Insert the generated blocks into the forest
-            for block in blocks {
-                forest.add_node(block).expect("Failed to add block");
+// ---------- Arbitrary implementations for missing types ----------
+
+impl Arbitrary for DebugOptions {
+    type Parameters = ();
+    type Strategy = BoxedStrategy<Self>;
+
+    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+        prop_oneof![
+            Just(DebugOptions::StackAll),
+            any::<u8>().prop_map(DebugOptions::StackTop),
+            Just(DebugOptions::MemAll),
+            (any::<u32>(), any::<u32>())
+                .prop_map(|(start, end)| DebugOptions::MemInterval(start, end)),
+            (any::<u16>(), any::<u16>(), any::<u16>()).prop_map(|(start, end, num_locals)| {
+                DebugOptions::LocalInterval(start, end, num_locals)
+            }),
+            any::<u16>().prop_map(DebugOptions::AdvStackTop),
+        ]
+        .boxed()
+    }
+}
+
+impl Arbitrary for AssemblyOp {
+    type Parameters = ();
+    type Strategy = BoxedStrategy<Self>;
+
+    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+        (
+            any::<bool>(),
+            prop::collection::vec(any::<char>(), 1..=20)
```

Comment:
There are better primitives for generating strings in `proptest` IMO - but we should ensure that the chars are at least printable, e.g. using `is_ascii_graphic` or generating from a specific range (see how we do it for `Ident` for example).